components/clm/src/biogeophys/SnowHydrologyMod.F90

```diff
@@ -587,24 +623,46 @@ subroutine SnowCompaction(bounds, num_snowc, filter_snowc, &
 
        ! Begin calculation - note that the following column loops are only invoked if snl(c) < 0
 
-       burden(bounds%begc : bounds%endc) = 0._r8
+       if (use_extrasnowlayers) then
+          do fc = 1, num_snowc
+             c = filter_snowc(fc)
+             burden(c) = 0._r8
+             zpseudo(c) = 0._r8
+             mobile(c) = .true.
+          end do
+       else
+          burden(bounds%begc : bounds%endc) = 0._r8
+       end if
 
        do j = -nlevsno+1, 0
           do fc = 1, num_snowc
              c = filter_snowc(fc)
+             if (use_extrasnowlayers) then
+                t = col_pp%topounit(c)
+             end if
              if (j >= snl(c)+1) then
 
-                wx = h2osoi_ice(c,j) + h2osoi_liq(c,j)
-                void = 1._r8 - (h2osoi_ice(c,j)/denice + h2osoi_liq(c,j)/denh2o) / dz(c,j)
+                if (.not. use_extrasnowlayers) then
+                   ! ++ams this is redundant and should be removed
+                   wx = h2osoi_ice(c,j) + h2osoi_liq(c,j)
+                   void = 1._r8 - (h2osoi_ice(c,j)/denice + h2osoi_liq(c,j)/denh2o) / dz(c,j)
+                   ! ams++
+                end if
+                
                 wx = (h2osoi_ice(c,j) + h2osoi_liq(c,j))
                 void = 1._r8 - (h2osoi_ice(c,j)/denice + h2osoi_liq(c,j)/denh2o)&
                      /(frac_sno(c) * dz(c,j))
                 ! If void is negative, then increase dz such that void = 0.
                 ! This should be done for any landunit, but for now is done only for glacier_mec 1andunits.
-                l = col_pp%landunit(c)
-                if (ltype(l)==istice_mec .and. void < 0._r8) then
-                   dz(c,j) = h2osoi_ice(c,j)/denice + h2osoi_liq(c,j)/denh2o
-                   void = 0._r8
+                
+                if (.not. use_extrasnowlayers) then
+                   ! ++ams I don't think this is necessary (removed in CLMv5)
```

Comment:
I'm not sure about this block.  I don't believe it (lines 660--663) is necessary, however, I'm not sure if removing it will maintain the model's BFB reproducibility.  Therefore I have "removed" these lines only for the new (use_extrasnowlayers) feature.